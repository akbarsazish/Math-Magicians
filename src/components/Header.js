import { Link } from 'react-router-dom';

const Header = () => (
  <div className="header">
    <div className="left">
      <h1 className="bigTitle"> Math Magician </h1>
    </div>
    <div className="right">
      <nav className="nav">
        <Link className="navItem" to="/"> Home </Link>
        <Link className="navItem" to="/calculator"> Calculator </Link>
        <Link className="navItem" to="/quots"> Qoute </Link>
      </nav>
    </div>
  </div>
);
export default Header;
