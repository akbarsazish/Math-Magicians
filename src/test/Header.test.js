import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Header from '../components/Header';

test('Header displays the correct title', () => {
  render(
    <Router>
      <Header />
    </Router>
  );

  const titleElement = screen.getByText('Math Magician');
  expect(titleElement).toBeInTheDocument();
});

test('Header displays the correct navigation links', () => {
  render(
    <Router>
      <Header />
    </Router>
  );

  const homeLink = screen.getByText('Home');
  const calculatorLink = screen.getByText('Calculator');
  const quoteLink = screen.getByText('Qoute'); // Note the typo in the component (should be "Quote")

  expect(homeLink).toBeInTheDocument();
  expect(calculatorLink).toBeInTheDocument();
  expect(quoteLink).toBeInTheDocument();
});

test('Header navigation links have the correct URLs', () => {
  render(
    <Router>
      <Header />
    </Router>
  );

  const homeLink = screen.getByText('Home');
  const calculatorLink = screen.getByText('Calculator');
  const quoteLink = screen.getByText('Qoute'); // Note the typo in the component (should be "Quote")

  expect(homeLink.getAttribute('href')).toBe('/');
  expect(calculatorLink.getAttribute('href')).toBe('/calculator');
  expect(quoteLink.getAttribute('href')).toBe('/quots'); // Note the typo in the component (should be "/quotes")
});
