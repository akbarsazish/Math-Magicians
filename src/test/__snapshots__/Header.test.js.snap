// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header displays the correct navigation links 1`] = `
<a
  class="navItem"
  href="/quots"
>
   Qoute 
</a>
`;

exports[`Header displays the correct title 1`] = `
<h1
  class="bigTitle"
>
   Math Magician 
</h1>
`;

exports[`Header navigation links have the correct URLs 1`] = `"/quots"`;
